pkg/restore/conflict_test.go

```diff
@@ -0,0 +1,216 @@
+/*
+Copyright 2018 the Heptio Ark contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package restore
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var result *unstructured.Unstructured
+
+var basicCase = [...]*unstructured.Unstructured{
+	unstructuredOrDie(
+		`
+		{
+			"apiVersion": "v1",
+			"kind": "ServiceAccount",
+			"metadata": {
+				"namespace": "ns1",
+				"name": "default"
+			},
+			"secrets": [
+				{ "name": "default-token-abcde" }
+			]
+		}
+		`,
+	),
+
+	unstructuredOrDie(
+		`
+		{
+			"kind": "ServiceAccount",
+			"apiVersion": "v1",
+			"metadata": {
+				"namespace": "ns1",
+				"name": "default"
+			},
+			"secrets": [
+				{ "name": "default-token-xzy12" }
+			]
+		}
+		`,
+	),
+}
+
+func BenchmarkMergeServiceAccountUnstructuredBasic(b *testing.B) {
+	var desired *unstructured.Unstructured
+
+	for n := 0; n < b.N; n++ {
+		desired, _ = mergeServiceAccountUnstructured(basicCase[0], basicCase[1])
+	}
+
+	result = desired
+}
+
+func BenchmarkMergeServiceAccountBasic(b *testing.B) {
+	var desired *unstructured.Unstructured
+
+	for n := 0; n < b.N; n++ {
+		desired, _ = mergeServiceAccounts(basicCase[0], basicCase[1])
+	}
+
+	result = desired
+}
+
+var multipleSecrets = [...]*unstructured.Unstructured{
+	unstructuredOrDie(
```

Comment:
Please add a comment that this is from-backup or in-cluster, and the same for the 2nd element